File: 2020/2020W20/preppindataw20.R
library(readxl)

xlsx <- "E:/2020W20 Input.xlsx"
decoder <- read_excel(xlsx, sheet = "Cipher") 
cipher_text <- read_excel(xlsx, sheet = "Encrypted Message")['Encrypted Message'][[1]]

plain_text <- chartr(paste(decoder[,2]), paste(decoder[,1]), cipher_text)
plain_text

File: 2020/2020W20/PreppindataW20.py
import pandas as pd

xlsx = pd.ExcelFile("E:/2020W20 Input.xlsx")
decoder = pd.read_excel(xlsx,'Cipher').set_index('Cipher').T.to_dict('records')[0]
#encoder = pd.read_excel(xlsx,'Cipher').set_index('Alphabet').T.to_dict('records')[0]

cipher_text = pd.read_excel(xlsx,'Encrypted Message')['Encrypted Message'][0]
plain_text = ''.join([decoder.get(c,c) for c in list(cipher_text)])
print(plain_text)
